### PR TITLE
[DS-4478] Optionally list all values of an array property.

### PR DESCRIPTION
**Pull request by mwoodiupui**
## References
https://jira.lyrasis.org/browse/DS-4478
#7813

## Description
Add an '--all' option which causes the command to list all values of an array property.

## Instructions for Reviewers
List of changes in this PR:
* Treat all properties as array properties.  Single-valued properties will be fetched as one-element arrays.
* Iterate over all values returned, displaying each on a separate line.
* Tidied up a few IDE suggestions:  private constructor, some brace formatting.

Testing:
`bin/dspace dsprop --all -p SOME_PROPERTY` should list zero values (a blank line) for a nonexistent property, one value for a single-valued property, and all configured values for an array property.

`bin/dspace dsprop --all --raw -p SOME_PROPERTY` should still display "raw" values (without property substitution) in all cases.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide)
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests for any bug fixes, improvements or new features.
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/master/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Review by pnbecker — APPROVED** [review]
Highly appreciated change. Great tests. Haven't tested it, but reviewed the code. +1

Thank you @mwoodiupui!

File: dspace-api/src/test/java/org/dspace/app/util/ConfigurationIT.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        argv = new String[] {
+            "--all",
+            "--property", MISSING_PROPERTY
+        };
+        expectedSystemExit.expectSystemExitWithStatus(0);
+        expectedSystemExit.checkAssertionAfterwards(new Assertion() {
+            @Override public void checkAssertion() {
+                String outputs = systemOutRule.getLogWithNormalizedLineSeparator();
+                String[] output = outputs.split("\n");
+                assertThat(output, arrayWithSize(0)); // Huh?  Shouldn't split() return { "" } ?
```
Comment (kosarko) [inline dspace-api/src/test/java/org/dspace/app/util/ConfigurationIT.java:226]:
```
jshell> String wtf = new String();
wtf ==> ""

jshell> String[] output = wtf.split("\n")
output ==> String[1] { "" }

jshell> String wtf = "\n";
wtf ==> "\n"

jshell> String[] output = wtf.split("\n")
output ==> String[0] {  }
```
So `outputs` is probably not an empty string, rather it's an empty line

**Comment by kosarko** [comment]
Haven't run it, looked at the code, seems ok. 

Are there any rules for licenses of dependencies added in `pom.xml`?

**Comment by mwoodiupui** [comment]
@kosarko Thank you for asking about dependency licensing.  We were already using Hamcrest, so the only totally new dependency seems to be `system-rules`.  That is distributed under the Common Public License, which is specifically listed as an example of an acceptable license in the DSpace [Code Contribution Guidelines](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions).

**Comment by tdonohue** [comment]
@mwoodiupui : Could we port this to `main` (for 7.x)?  This looks to have had approval for 6.4, but it'd be important to ensure this is ported to `main` as well.

**Comment by mwoodiupui** [comment]
This was ported to `main` as #3312 and merged on 24-Mar-2022.

**Comment by tdonohue** [comment]
@mwoodiupui : The version that was ported/merged into main is different from this one.  If you recall, the `main` version has a `--first` flag (by default displaying all values of an array).  This PR has an `--all` flag (by default displaying the first value of an array).

So, this PR would need to be updated.  Or, alternatively we could just close this PR & say it's "fixed in 7.3".

**Comment by tdonohue** [comment]
Closing, fixed in 7.3 by #3312

**Pull request closed without merging**